src/platform/utilities/ui/index.js

```diff
@@ -84,3 +85,12 @@ export function scrollAndFocus(errorEl) {
 export function displayPercent(decimalNumber, places = 0) {
   return `${(decimalNumber * 100).toFixed(places)}%`;
 }
+
+export function isReactComponent(value) {
```

Comment:
This is the core new check. It's more complicated than the one in RJSF because you'd never pass React elements directly as `ui:field` or `ui:widget` (you always need to use the props), but we do have cases for that in the form system (like the description field).